components/movementsensor/merged/merged.go

```diff
@@ -0,0 +1,373 @@
+
+type merged struct {
+	resource.Named
+	logger golog.Logger
+
+	mu sync.Mutex
+
+	ori     movementsensor.MovementSensor
+	pos     movementsensor.MovementSensor
+	compass movementsensor.MovementSensor
+	linVel  movementsensor.MovementSensor
+	angVel  movementsensor.MovementSensor
+	linAcc  movementsensor.MovementSensor
+
+	oriReporting     bool
+	posReporting     bool
+	compassReporting bool
+	linVelReporting  bool
+	angVelReporting  bool
+	linAccReporting  bool
+}
+
+func init() {
+	resource.Register(movementsensor.API, model,
+		resource.Registration[movementsensor.MovementSensor, *Config]{
+			Constructor: newMergedModel,
+		})
+}
+
+func newMergedModel(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger golog.Logger) (
+	movementsensor.MovementSensor, error,
+) {
+	m := merged{
+		logger: logger,
+	}
+
+	if err := m.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+
+	return &m, nil
+}
+
+func (m *merged) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
+	newConf, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return err
+	}
+
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	// check if we need to reconfigure, we only return the first good sensor in this driver
+	// so we reconfigure each type of senser based on this comparison
+	sensorChanged := func(cfgVals []string, ms movementsensor.MovementSensor) bool {
+		if ms == nil && len(cfgVals) != 0 {
+			return true
+		}
+
+		if ms != nil && len(cfgVals) != 0 {
+			// check if the new first entry of the list of sensors is
+			// names the same as the sesnor currently used by this model
+			return (cfgVals[0] != ms.Name().ShortName())
+		}
+
+		return false
+	}
+
+	oriChanged := sensorChanged(newConf.Orientation, m.ori)
+	posChanged := sensorChanged(newConf.Position, m.pos)
+	compassChanged := sensorChanged(newConf.CompassHeading, m.compass)
+	lvelChanged := sensorChanged(newConf.LinearVelocity, m.linVel)
+	avelChanged := sensorChanged(newConf.AngularVelocity, m.angVel)
+	laccChanged := sensorChanged(newConf.LinearAcceleration, m.linAcc)
+
+	firstGoodSensorWithProperties := func(deps resource.Dependencies, names []string, mprops *movementsensor.Properties,
+	) (movementsensor.MovementSensor, error) {
+		// check if the config names and dependencies have been passed at all
+		if len(names) == 0 || deps == nil {
+			return nil, nil
+		}
+
+		var ms movementsensor.MovementSensor
+		for _, name := range names {
+			depsms, err := movementsensor.FromDependencies(deps, name)
+			if err == nil {
+				ms = depsms
+				// found the first good sensor for this information
+				// don't check the rest
+				break
+			} else {
+				continue
+			}
+		}
+
+		props, err := ms.Properties(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		if reflect.DeepEqual(props, mprops) {
+			return ms, nil
+		}
+		return nil, fmt.Errorf("none of the sensors in provided list %#v statisfy properties, need %#v", names, mprops)
```

Comment:
I expect this error to read "...satisfy properties, need {LinearVelocitySupported: false, AngularVelocitySupported: false, OrientationSupported: false, ...}" Is there an easy way to remove the noise and only list the one property that we're checking here, so it's easy to see what the error is complaining about? 

This isn't a _great_ solution, but the best I see right now is to pass another argument to this function, which is a string containing the property name that should be mentioned here. (which is not a great solution because it's easy to copy and paste so that the property name doesn't actually match the property being checked, which is one advantage your current solution has)